### PR TITLE
Pyarrow table

### PR DESCRIPTION
**Pull request by perolavsvendsen**
Adding support for pyarrow tables and the smry content

### CODE REVIEW

File: src/fmu/dataio/_export_item.py [hunk 1]
```diff
@@ -49,6 +52,7 @@
     "volumes": None,
     "volumetrics": None,  # or?
     "khproduct": None,
+    "smry": None,
 }
```
Comment (jcrivenaes) [inline src/fmu/dataio/_export_item.py:57]: Better with 'summary'? smry is Eclipse delimited language (aka old MS DOS)

File: tests/test_fmu_dataio_table.py [hunk 2]
```diff
@@ -30,30 +31,55 @@
...
+
+    # make a small pa.Table
+    df = pd.DataFrame({"STOIIP": [123, 345, 654], "PORO": [0.2, 0.4, 0.3]})
+    table = pa.Table.from_pandas(df)
+    fmu.dataio.ExportData.export_root = tmp_path.resolve()
+
+    exp = fmu.dataio.ExportData(name="test", verbosity="INFO", content="smry")
+    exp._pwd = tmp_path
+    exp.to_file(table)
+
```
Comment (jcrivenaes) [inline tests/test_fmu_dataio_table.py:69]: Instead of setting `exp._pwd`, I think is better to set `ExportData(... runfolder=tmp_path)` since `self._pwd` is actually used in the __init__.

File: src/fmu/dataio/_export_item.py (hunk 1 above)
Comment (perolavsvendsen) [reply src/fmu/dataio/_export_item.py:57]: @berland @anders-kiaer @asnyv any strong feelings?

Comment (anders-kiaer) [reply src/fmu/dataio/_export_item.py:57]: No strong opinions, I'm fine with both. If I should try find an argument for `smry` it is because that particular abbrevaition is so arcane/strange looking that it can't be interpreted as anything else than Eclipse/Flow simulation output. The word "Summary" could mean something else for non-reservoir-engineers perhaps and/or is more general?

TL;DR: No strong feelings. I'm fine with both (I would have preferred something else than smry/summary was chosen back in the days, but that 🚋 left when at preschool or something)

Comment (asnyv) [reply src/fmu/dataio/_export_item.py:57]: I was writing pretty much the same thing as @anders-kiaer, but he beat me to it... 😅 I agree that the likelyhood of mis-interpretation is less with `smry`. 
Another thing: what about other time-dependent data which is not Eclipse/Flow output? The simple example: output of postprocessing/calculations upon `smry` output

File: tests/test_fmu_dataio_table.py (hunk 2 above)
Comment (perolavsvendsen) [reply tests/test_fmu_dataio_table.py]: Done ✔

File: src/fmu/dataio/_export_item.py (hunk 1 above)
Comment (jcrivenaes) [reply src/fmu/dataio/_export_item.py:57]: I have a better feeling with `summary` is this is a word ;-)

**Review by jcrivenaes — APPROVED** [review]
Good to go

File: src/fmu/dataio/_export_item.py (hunk 1 above)
Comment (berland) [reply src/fmu/dataio/_export_item.py:57]: I once changed from `ecl2csv smry` to `ecl2csv summary` as I got uncomfortable by the short-hand word (and also for consistency reasons). What about more drastic other options, `timeseries`? "summary" is a very misleading word except for the choir.

Comment (perolavsvendsen) [reply src/fmu/dataio/_export_item.py:57]: I agree with @berland. "summary" could mean a lot of things, even in an FMU context, which may or may not be related to `smry`. So if we are going away from `smry` I would also suggest going a bit further than just expanding it. `timeseries` sounds very good to me. That's the word I normally use. But I don't know (and don't care 😐) how much confusion it may or may cause deep into the res.eng domain. Another option could be `vectors` I guess. But I like `timeseries` better, I think.

Comment (perolavsvendsen) [reply src/fmu/dataio/_export_item.py:57]: An argument for keeping "smry" (or similar) would be in the case of needing specific inputs relating only to smry-data. The current pattern is that contents supporting/requiring additional metadata is handled through allowing a tag with the same name as the content, e.g.

```
{
  content: mycontent,
  mycontent: {
    someattribute: x
  }
}
```

If `timeseries` is a content, we can probably think of many different use cases inside FMU, which may or may not require different additional metadata contents. With that perspective, being specific to smry data is perhaps smart(er)?

Comment (jcrivenaes) [reply src/fmu/dataio/_export_item.py:57]: I will still downvote `smry` as it is just an arcane abbreviation, belonging to RE tribal language. Here is the current list of allowed content:
```
ALLOWED_CONTENTS = {
    "depth": None,
    "time": None,
    "thickness": None,
    "property": {"attribute": str, "is_discrete": bool},
    "seismic": {
        "attribute": str,
        "zrange": float,
        "filter_size": float,
        "scaling_factor": float,
    },
    "fluid_contact": {"contact": str},
    "field_outline": {"contact": str},
    "regions": None,
    "pinchout": None,
    "subcrop": None,
    "fault_lines": None,
    "velocity": None,
    "volumes": None,
    "volumetrics": None,  # or?
    "khproduct": None,
    "smry": None,
}
```
All others are more or less spoken words. The summary data here we deal with are time series for various well vectors, right? If `timeseries` is too wide, So what about `flow_timeseries` or `well_timeseries`?

Comment (anders-kiaer) [reply src/fmu/dataio/_export_item.py:57]: Is it possible to specify metadata regarding what is the immediate "parent" tool that created data in sumo? E.g. Eclipse or Flow in this case (could perhaps even include version of the software tool as metadata for documentation?).

What we discuss here is the "easy" case of simulation time series output from a simulator. As @asnyv mentions above: What about when these series are combined to create new time series during postprocessing? Are they still `x in [summary, smry]`, or are they something else? Does it depend on the complexity of the calculation ("+, -, *, /" only, vs. arbitrary `f(x)` functions using other input data as well in the calculation like compressor tables for calculating time series of associated CO₂ emissions)?

If `timeseries` + some metadata on the tool that created the output in question is on the table, I think my vote would go in that direction. Both `summary` and `smry` are quite res.eng-oriented terminology, and it raises the next question on when is derived data from `smry/summary` not `smry/summary` anymore.

Comment (anders-kiaer) [reply src/fmu/dataio/_export_item.py:57]: > So what about `flow_timeseries` or `well_timeseries`?

It might be difficult to find a short and good prefix. E.g. the typical set of output time series from Flow/Eclipse contains more than time series associated with wells, and also vectors associated with other things than `flow` (at least in the direct meaning of the word `flow`). `reservoir_simulator_timeseries` is perhaps too long?

Comment (berland) [reply src/fmu/dataio/_export_item.py:57]: My vote is for `timeseries`, and then add metadata to specify source.

Comment (berland) [reply src/fmu/dataio/_export_item.py:57]: (this is perhaps relevant to peek at: https://resqpy.readthedocs.io/en/latest/_autosummary/resqpy.time_series.html#module-resqpy.time_series )

Comment (perolavsvendsen) [reply src/fmu/dataio/_export_item.py:57]: Renamed to `timeseries` now. Probably just pushing the problem further in now, but what should the associated metadata (as a minimum) contain?

```
timeseries: {
  source: "smry"
}
```
?

Comment (anders-kiaer) [reply src/fmu/dataio/_export_item.py]: What about something more direct towards the tool writing the output, e.g. `source: {software: "Eclipse", version: "2021.1"}` (just as an example, not sure if this is a good metadata dictionary for sumo).

I think if we go for `timeseries` we should use the opportunity of not mentioning `smry`. Today Eclipse/Flow time series is usually read from `.UNSMRY` files, but the same simulators I think can write the same data to `HDF5` (and maybe `.arrow` in the future). Same data from the same tool, just written in different output file formats.

Comment (perolavsvendsen) [reply src/fmu/dataio/_export_item.py]: A common and fairly fundamental problem is that data like that must be manually entered. In the "original" metadata used by JS, software and version was specified both for RMS and Eclipse. But it was manually entered, so not very sustainable and very, very error-prone. Same goes for `workflow` which cannot be extracted - but needs to be included, as it in many cases is the only difference between two data objects being exported (e.g. you export a horizon representation 4-5-6 times during the workflow to monitor the changes in each step, etc).

Technically this is not a Sumo issue at all, it is pure FMU...
